clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -325,6 +326,13 @@ static std::string getModuleContextHash(const ModuleDeps &MD,
   // will be readable.
   HashBuilder.add(getClangFullRepositoryVersion());
   HashBuilder.add(serialization::VERSION_MAJOR, serialization::VERSION_MINOR);
+  if (CI.getFileSystemOpts().WorkingDir.empty()) {
+    llvm::ErrorOr<std::string> CWD = VFS.getCurrentWorkingDirectory();
+    if (CWD)
+      HashBuilder.add(*CWD);
+  }
+  // If any of the above options are set, then there must have been a command
```

Comment:
Sometime options are implicitly set in the driver/toolchain based on other external factors. Wouldn't it be more reliable to always hash the WorkingDir?

Reply:
This is for -cc1 so it's always explicit, but it's fine to include it anyway.